src/libraries/Microsoft.Extensions.Http/src/DefaultHttpClientFactory.cs

```diff
@@ -343,24 +393,48 @@ public static class EventIds
                 "HttpMessageHandler expired after {HandlerLifetime}ms for client '{ClientName}'");
 
 
-            public static void CleanupCycleStart(ILogger logger, int initialCount)
+            public static void CleanupCycleStart(Lazy<ILogger> loggerLazy, int initialCount)
+            {
+                if (TryGetLogger(loggerLazy, out ILogger? logger))
+                {
+                    _cleanupCycleStart(logger, initialCount, null);
+                }
+            }
+
+            public static void CleanupCycleEnd(Lazy<ILogger> loggerLazy, TimeSpan duration, int disposedCount, int finalCount)
             {
-                _cleanupCycleStart(logger, initialCount, null);
+                if (TryGetLogger(loggerLazy, out ILogger? logger))
+                {
+                    _cleanupCycleEnd(logger, duration.TotalMilliseconds, disposedCount, finalCount, null);
+                }
             }
 
-            public static void CleanupCycleEnd(ILogger logger, TimeSpan duration, int disposedCount, int finalCount)
+            public static void CleanupItemFailed(Lazy<ILogger> loggerLazy, string clientName, Exception exception)
             {
-                _cleanupCycleEnd(logger, duration.TotalMilliseconds, disposedCount, finalCount, null);
+                if (TryGetLogger(loggerLazy, out ILogger? logger))
+                {
+                    _cleanupItemFailed(logger, clientName, exception);
+                }
             }
 
-            public static void CleanupItemFailed(ILogger logger, string clientName, Exception exception)
+            public static void HandlerExpired(Lazy<ILogger> loggerLazy, string clientName, TimeSpan lifetime)
             {
-                _cleanupItemFailed(logger, clientName, exception);
+                if (TryGetLogger(loggerLazy, out ILogger? logger))
+                {
+                    _handlerExpired(logger, lifetime.TotalMilliseconds, clientName, null);
+                }
             }
 
-            public static void HandlerExpired(ILogger logger, string clientName, TimeSpan lifetime)
+            private static bool TryGetLogger(Lazy<ILogger> loggerLazy, [NotNullWhen(true)] out ILogger? logger)
             {
-                _handlerExpired(logger, lifetime.TotalMilliseconds, clientName, null);
+                logger = null;
+                try
+                {
+                    logger = loggerLazy.Value;
+                }
+                catch { } // not throwing in logs
```

Comment:
This eats all exceptions?  What was being done before?

Reply:
(now I believe it is also needed in case the factory was never disposed...)

Reply:
Maybe we should only eat the documented exceptions for [`Lazy<T>.Value`](https://learn.microsoft.com/en-us/dotnet/api/system.lazy-1.value?view=net-7.0) which are:

-  [MemberAccessException](https://learn.microsoft.com/en-us/dotnet/api/system.memberaccessexception?view=net-7.0)
- [MissingMemberException](https://learn.microsoft.com/en-us/dotnet/api/system.missingmemberexception?view=net-7.0)
- [InvalidOperationException](https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-7.0)

I'm not sure _any_ of those would actually throw, so perhaps the only likely exception would be null access?

Reply:
The actual reported one is `ObjectDisposedException`; though, it could be _any_ exception thrown by e.g. a 3rd party `ILoggerProvider`/`ILoggerProvider`/failing to create `ILogger` instance because of some other problem.

Reply:
Theoretically I _can_ only swallow ObjectDisposedException, that would fix the reported issue. But if we ever stumble upon some other ILogger creation exception, this will become a regression, since it would crash the process where it was not crashing before.

Reply:
(I meant not that the current PR would introduce a regression, but this one did https://github.com/dotnet/runtime/pull/89531)